From 26fbaf917c8349316e0213072529cb1598f05f9a Mon Sep 17 00:00:00 2001
From: Iris Chang <iris.chang@mediatek.com>
Date: Sun, 1 Jul 2018 23:29:11 +0800
Subject: [PATCH 5/6] DO NOT MERGE: Fix landscape LCM issue

When device uses landscape LCM, the nature of the screen shows that
the device is a portrait device. In this case, we find GPU filter
function is opened by mistake when capturing screen and CTS case
fails because image pixel value is changed.

The solution is to correct the GPU filter by hworientation.

Bug: 69691076
Test:
1. Capture screen
2. android.uirendering.cts.testclasses.SurfaceViewTests#testMovingWhiteSurfaceView

Change-Id: I67b75fc8fab188d24f1d5febff20d9bd10c15204
---
 services/surfaceflinger/SurfaceFlinger.cpp | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/services/surfaceflinger/SurfaceFlinger.cpp b/services/surfaceflinger/SurfaceFlinger.cpp
index 2dd3078e1..ee68a6731 100644
--- a/services/surfaceflinger/SurfaceFlinger.cpp
+++ b/services/surfaceflinger/SurfaceFlinger.cpp
@@ -4392,8 +4392,14 @@ void SurfaceFlinger::renderScreenImplLocked(
     // get screen geometry
     const int32_t hw_w = hw->getWidth();
     const int32_t hw_h = hw->getHeight();
-    const bool filtering = static_cast<int32_t>(reqWidth) != hw_w ||
-                           static_cast<int32_t>(reqHeight) != hw_h;
+    bool filtering = false;
+    if (mHwOrientation & DisplayState::eOrientationSwapMask) {
+        filtering = static_cast<int32_t>(reqWidth) != hw_h ||
+                static_cast<int32_t>(reqHeight) != hw_w;
+    } else {
+        filtering = static_cast<int32_t>(reqWidth) != hw_w ||
+                static_cast<int32_t>(reqHeight) != hw_h;
+    }
 
     // if a default or invalid sourceCrop is passed in, set reasonable values
     if (sourceCrop.width() == 0 || sourceCrop.height() == 0 ||
-- 
2.17.1

